File: weekly-contest-222/python/solB.py
import sys
import argparse

class Solution:
    def countPairs(self, deliciousness: List[int]) -> int:
        f = {}
        n = len(deliciousness)
        ans = 0
        for i in range(0,n):
            for j in range(0, 41):
                u = 1<<j
                if u-deliciousness[i] in f:
                    ans += f[u-deliciousness[i]]
                    ans %= 1000000007
            f[deliciousness[i]] = 1 if deliciousness[i] not in f else f[deliciousness[i]]  + 1 
        return ans

def main():
    pass

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument('--file', dest='filename', default=None)
    args = parser.parse_args()
    if args.filename is not None:
        sys.stdin = open(args.filename)
    main()

File: weekly-contest-222/cpp/solB.cpp
#include <iostream>
#include <vector>
#include <string>
#include <map>

using namespace std;
#define MODN 1000000007
typedef long long ll;
class Solution {
public:
    int countPairs(vector<int>& deliciousness) {
        map<ll,ll> f;
        int n = deliciousness.size();
        ll ans = 0;
        for(int i = 0; i < n; ++i){
            for(ll j = 0; j < 41; ++j){
                ll u = 1LL << j;
                ans += f[u-deliciousness[i]];
                ans %=MODN;
            }
            ++f[deliciousness[i]];
        }
        return ans;
    }
};

int main()
{
    
    #ifndef ONLINE_JUDGE
        freopen("input.txt","r",stdin);
        freopen("output.txt","w",stdout);
    #endif
    return 0;
}